### PR TITLE
gh-114075: Capture `test_compileall` output 

### PR DESCRIPTION
**Pull request by Eclips4**
* Issue: gh-114075

### CODE REVIEW

**Review by AlexWaygood — COMMENTED** [review]
Consider using `test.support.captured_stdout` (it's just a convenience wrapper that slightly simplifies what you already have)

**Comment by Eclips4** [comment]
> Consider using `test.support.captured_stdout` (it's just a convenience wrapper that slightly simplifies what you already have)

Thanks!

Also, there is no need in backports, I cannot reproduce it on 3.11 & 3.12 branches.

File: Lib/test/test_compileall.py [hunk 1]
```diff
@@ -369,7 +369,8 @@ def test_strip_only_invalid(self):
         script = script_helper.make_script(path, "test", "1 / 0")
         bc = importlib.util.cache_from_source(script)
         stripdir = os.path.join(self.directory, *(fullpath[:2] + ['fake']))
-        compileall.compile_dir(path, quiet=True, stripdir=stripdir)
+        with support.captured_stdout():
+            compileall.compile_dir(path, quiet=True, stripdir=stripdir)
```
Comment (AlexWaygood) [inline Lib/test/test_compileall.py]:
```suggestion
        with support.captured_stdout() as out:
            compileall.compile_dir(path, quiet=True, stripdir=stripdir)
        self.assertIn("not a valid prefix", out.getvalue())
```

**Review by serhiy-storchaka — COMMENTED** [review]
The output was added in gh-105931 (PR #108671), when this test was also added.

It can be simply silenced by passing `quiet=2`. But since there is no other test for this output, it is better to capture the stdout, and then test that an expected message occurred in the captured output. Please add this, not simply ignore it.

**Review by AlexWaygood — APPROVED** [review]
LGTM. @serhiy-storchaka, I think @Eclips4's latest push addresses your remark

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by Eclips4** [comment]
Thanks for your reviews!
🚀

**Pull request merged**